components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/filter/AuthorizationHeaderFilter.java

```diff
@@ -0,0 +1,146 @@
+        final HttpServletRequest req = (HttpServletRequest) request;
+        final String authHeader = req.getHeader(HTTPConstants.HEADER_AUTHORIZATION);
+        if (StringUtils.isEmpty(authHeader)) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        String authType = null;
+        if (authHeader.length() >= AUTH_TYPE_BASIC_LENGTH) {
+            authType = authHeader.trim().substring(0, AUTH_TYPE_BASIC_LENGTH);
+        }
+        if (AUTH_TYPE_BASIC.equals(authType)) {
+            final String authCredentials = (authHeader.trim().substring(AUTH_TYPE_BASIC_LENGTH)).trim();
+            //if auth header comes in invalid format send error in response
+            if (StringUtils.isBlank(authCredentials) || authCredentials.indexOf(' ') >= 0
+                    || authCredentials.length() < MINIMUM_CREDENTIAL_SIZE) {
+                String errorMsg = "Invalid credentials";
+                handleErrorResponse((HttpServletResponse) response, HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+                        errorMsg);
+            } else if (authHeader.substring((AUTH_TYPE_BASIC + " ").length()).startsWith(" ")) {
+                //if there is more than single space between auth_type and credentials modify the request header
+                HttpServletRequestWrapper wrapper = new HttpServletRequestWrapper(req) {
+                    @Override
+                    public Enumeration<String> getHeaders(String name) {
+                        if (StringUtils.equalsIgnoreCase(name, HTTPConstants.HEADER_AUTHORIZATION)) {
+                            Enumeration<String> headerValues = req.getHeaders(name);
+                            ArrayList<String> newHeaderValues = new ArrayList<>();
+                            while (headerValues.hasMoreElements()) {
+                                String value = headerValues.nextElement();
+                                if (StringUtils.equals(value, authHeader)) {
+                                    value = AUTH_TYPE_BASIC + " " + authCredentials;
+                                    newHeaderValues.add(value);
+                                } else {
+                                    newHeaderValues.add(value);
+                                }
+                            }
+                            return Collections.enumeration(newHeaderValues);
+                        }
+                        return super.getHeaders(name);
+                    }
+                };
+                // pass the request along the filter chain
+                chain.doFilter(wrapper, response);
+            }
+            //if auth header comes in correct format, forward
+            else {
+                chain.doFilter(request, response);
+            }
+        } else {
+            chain.doFilter(request, response);
+        }
+    }
+
+    /**
+     * Clean up any resources being held by the filter
+     */
+    @Override
+    public void destroy() {
```

Comment:
This method is called by the web container once all the threads in the doFilter method have exited indicating to the the filter class that it is being taken out of service. So we can use the implementation of this method to clean up any resources if they are being held by the filter.